seed/tasks.py

```diff
@@ -322,49 +323,50 @@ def _finish_delete_cycle(cycle_id, prog_key):
 @lock_and_track
 def delete_organization_column(column_pk, org_pk, prog_key=None, chunk_size=100, *args, **kwargs):
     """Deletes an extra_data column from all merged property/taxlot states."""
-
-    column = Column.objects.get(id=column_pk, organization_id=org_pk)
-
     progress_data = ProgressData.from_key(prog_key) if prog_key else ProgressData(
         func_name='delete_organization_column', unique_id=column_pk)
 
+    tasks = []
+    tasks.append(
+        _generate_tasks_to_delete_organization_column.subtask((column_pk, org_pk, progress_data.key, chunk_size))
+    )
+
+    chord(tasks, interval=15)(_finish_delete_column.subtask([column_pk, progress_data.key]))
+    return progress_data.result()
+
+
+@shared_task
+def _generate_tasks_to_delete_organization_column(column_pk, org_pk, prog_key, chunk_size, *args, **kwargs):
+    """ Find -States with column to be deleted """
+    column = Column.objects.get(id=column_pk, organization_id=org_pk)
+
     ids = []
 
     if column.table_name == 'PropertyState':
-        ids = list(
-            PropertyState.objects.filter(organization_id=org_pk, data_state=DATA_STATE_MATCHING,
-                                         extra_data__has_key=column.column_name).values_list('id', flat=True)
-        )
+        ids = PropertyState.objects.filter(organization_id=org_pk, data_state=DATA_STATE_MATCHING,
+                                           extra_data__has_key=column.column_name).values_list('id', flat=True)
     elif column.table_name == 'TaxLotState':
-        ids = list(
-            TaxLotState.objects.filter(organization_id=org_pk, data_state=DATA_STATE_MATCHING,
-                                       extra_data__has_key=column.column_name).values_list('id', flat=True)
-        )
+        ids = TaxLotState.objects.filter(organization_id=org_pk, data_state=DATA_STATE_MATCHING,
+                                         extra_data__has_key=column.column_name).values_list('id', flat=True)
 
+    """ Evaluate a queryset in a task allowing web processes to complete """
+    progress_data = ProgressData.from_key(prog_key)
     total = len(ids)
-
-    # total is the number of records divided by the chunk size
     progress_data.total = total / float(chunk_size)
     progress_data.data['completed_records'] = 0
     progress_data.data['total_records'] = total
     progress_data.save()
 
-    tasks = []
```

Comment:
since we are already inside of a task, simply run it. No need to add to a task list

Reply:
I think the only benefit to creating a "task" list would be that we could parallelize the deletion, maybe making it `num_workers` times faster. Though I'm not that it matters too much here... I'm honestly fine keeping it how you have it. I would probably rename this function though since it's not actually generating any new tasks. Also, we should probably remote the `@shared_task` decorator on `_delete_organization_column_chunk` just so we know it's not actually used as a celery task.